### DISCUSSION TITLE
Birt Report Calling Problem from JSP

### DISCUSSION THREAD

**Discussion by irfankundi786** [eclipse-birt/birt#1677] [Q&A]
I have issues with calling Report from jsp i am using jdk 1.8 and Birt version Photon (4.8) Build id: I20180516-2000
i have done these steps but still having problem to call it.These are steps i already performed

1. i have copy birt.tld in Web-Inf\tlds folder
2. web.xml entry as
 ```
<jsp-config>
    <taglib>
        <taglib-uri>/birt.tld</taglib-uri>
        <taglib-location>/WEB-INF/tlds/birt.tld</taglib-location>
    </taglib>
</jsp-config>
```
3. copy the jar file in path but still not sure about which are the jar file must copy i have copied bundles of jar file Please Mention name of jar file which mandatory
4. copy birt-viewer in tomcat webapp folder and able to run those report in browser from birt-viewer copied folder as i copy one of report to this folder and able to run in browser as   http://localhost:8080/birt-viewer/frameset?__report=Employee.rptdesign
5. My jsp code is 
```
<%@ page language="java" contentType="text/html; charset=ISO-8859-1"
pageEncoding="ISO-8859-1"%>
<%@ taglib uri="/birt.tld" prefix="birt" %>
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html>
    <head>
        <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
        <title>JSP Page</title>
    </head>
    <body>
        <h2>Welcome to Birt Report Page</h2>
<birt:parameterPage id="birtParmPage" 
    reportDesign="http://localhost:8080/birt-viewer/frameset?__report=Employee.rptdesign"
    name="my form"
    pattern="frameset"
    height="600"
    width="800"
    format="html"
    title="My Viewer Tag"
    isCustom="true"
    showTitle="true"
    showToolBar="true"
    showNavigationBar="true">
    
    <input type="submit" value="Run Report">
</birt:parameterPage>

    </body>
</html>
``` 
Still facing problem

**Comment by hvbtup** [comment]
First of all, I can only see the JSP code if I edit your question. You did not properly quote it, so I edited your question.

Next, you only say "does not work". 
Please be more specific:
 * What did you expect?
 * What happens instead?
 * Are there any error messages in the browser or in the server logs?

I never used BIRT from JSP, but too me your JSP code looks wrong, in particular the `reportDesign` attribute value.
I think this should contain a file URI for the rptdesign file on the server, or just the file name without a path, but I could be wrong.

You should use your browsers developer settings too see how the birt viewer page works.

**Reply by irfankundi786** [reply]
my Jsp code is here 
`<%@ page language="java" contentType="text/html; charset=ISO-8859-1"
   pageEncoding="ISO-8859-1"%>
   <%@ taglib uri="/birt.tld" prefix="birt" %>
  <!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
    <html>
      <head>
        <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
        <title>JSP Page</title>
      </head>
    <body>
        <h2>Welcome to Birt Report Page</h2>
  <birt:viewer id="birtViewer" reportDesign="/WEB-INF/reports/Employee.rptdesign"  
                     pattern="run" height="600" width="1000" format="pdf" 
                     scrolling="yes" showParameterPage="false" >
    <birt:param name="param1" value='<%=request.getParameter("txtParam1")%>'></birt:param>
    <birt:param name="param2" value='<%=request.getParameter("txtParam1")%>'></birt:param>
</birt:viewer>

      </body>
      </html>
`
I place my report in \WEB-INF\reports folder and want to access it through <birt:viewer> as mention above but not able to show the report it gives error as not found (error code 404)
